client/dashboard/app/queries/domain-glue-records.ts

```diff
@@ -8,10 +14,47 @@ export const domainGlueRecordsQuery = ( domainName: string ) =>
 		queryFn: () => fetchDomainGlueRecords( domainName ),
 	} );
 
+export const domainGlueRecordCreateMutation = ( domainName: string ) =>
+	mutationOptions( {
+		mutationFn: ( glueRecord: DomainGlueRecord ) => createDomainGlueRecord( glueRecord ),
+		onSuccess: ( newData: unknown, createdGlueRecord: DomainGlueRecord ) => {
```

Comment:
Great updating the `queryClient` on success 👍 
Another heads up that we should not provide data types for the `onSuccess` function. The `mutationOptions()` helper makes sure these types are correct. The `newData` will automatically be whatever type `mutationFn` returns, and `createdGlueRecord` will automatically be whatever type `mutationFn` accepts.

I've found it is best to let TanStack Query infer the types where it can because it's actually easy to mistype them. When I first rolled out the `mutationOptions` and `queryOptions` helpers I found a couple of runtime bugs because of incorrect types 😁 


```suggestion
		onSuccess: ( newData, createdGlueRecord ) => {
```

Same advice for the other `onSuccess` handlers.

You do have to supply a type for the `mutationFn` parameter though. There's no way for TanStack to know what you want this type to be (but don't worry because TS will complain about it if you forget)

Reply:
Thanks for teaching me the ropes, Philip! 🙇

The types were [updated](https://github.com/Automattic/wp-calypso/pull/105213/commits/76dee31974bb26c6797ae0e44dbd802a65a2edab).